### DISCUSSION TITLE
"Service Unavailable" every couple days

### DISCUSSION THREAD

**Discussion by suntrop** [plausible/analytics#3306] [Self-Hosted Support]
I have been testing Plausible for ~8 weeks now. There are just 3 sites tracked (< 50 visits per day overall) in this instance.

About once a week, it goes offline. When I visit my Dashboard, I get:

```
Service Unavailable
The server is temporarily unable to service your request due to maintenance downtime or capacity problems. Please try again later.
```

I have to start the Docker container again, and it runs fine for another ~week. As said, there are not many visitors and my VPS is running well. The same VPS has Fathom and Umami running too, and there are no such problems. But those three are actually my first three Docker instances, I am using. Don't have much experience with Docker. I did "docker ps", however, and it looked like the container was still running. So I guess the problem is not Docker itself.

**Comment by ruslandoga** [comment]
👋 @suntrop 

Could you please provide more info about your setup? How do you deploy it, do you use `docker-compose.yml` from [plausible/hosting](https://github.com/plausible/hosting)? What kind of host are you running on, CPU, RAM, load average, etc.? Are there any logs for the failed `plausible/analytics` containers: `docker ps -a`, `docker logs <contianer-id>`, `docker inspect <container-id>`?

**Comment by suntrop** [comment]
ruslandoga, I greatly appreciate your help!

Sure, here is more info:

1. Yes, I followed the instructions from the docs. Didn't change anything, beside using Apache reverse proxy
2. Ubuntu 20.04, KVM, 4 GB RAM, CPU avg 0.16 (monitoring looks normal*)
3. Docker log: https://dsc.cloud/40757e/dockerlogfile.log
4. Inspect log: https://dsc.cloud/40757e/inspect.log

I only see times, but no dates inside the log file, so I don't know whether those are from recent incidents or from when I installed it. I can't read much out from this. Sorry.

* Although my monitoring appears to be normal, I think that most or all of the downtimes occurred on weekends. I perform backups every night, but on weekends I do an additional backup, which may require more resources. However, none of the other Docker containers have experienced any issues since they were installed in June.

**Reply by ruslandoga** [reply]
Do the DB connection error times correspond to when backups happen?

From the logs it seems like at some point one of the dependencies of Plausible (Oban, a job queue) gives up connecting to the database and causes the app to shutdown.

```
01:54:19.214 [error] GenServer {Oban.Registry, {Oban, Oban.Peer}} terminating
** (stop) exited in: GenServer.call(#PID<0.9120.1>, {:listen, [:leader]}, 5000)
    ** (EXIT) time out
    (elixir 1.13.4) lib/gen_server.ex:1030: GenServer.call/3
    (postgrex 0.16.4) lib/postgrex/simple_connection.ex:212: Postgrex.SimpleConnection.call/3
    (oban 2.12.0) lib/oban/peers/postgres.ex:75: Oban.Peers.Postgres.handle_continue/2
    (stdlib 3.17.1) gen_server.erl:695: :gen_server.try_dispatch/4
    (stdlib 3.17.1) gen_server.erl:437: :gen_server.loop/7
    (stdlib 3.17.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
{"Kernel pid terminated",application_controller,"{application_terminated,plausible,shutdown}"}
```

But then Docker fails to restart it since the default Docker Compose command tries to run the migrations before starting the container, and that's not possible while PostgreSQL is unavailable:

```
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist
01:54:41.161 [error] Postgrex.Protocol (#PID<0.173.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (plausible_db:5432): non-existing domain - :nxdomain
01:54:41.161 [error] Postgrex.Protocol (#PID<0.172.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (plausible_db:5432): non-existing domain - :nxdomain
01:54:41.140 [error] GenServer #PID<0.190.0> terminating
** (DBConnection.ConnectionError) tcp connect (plausible_db:5432): non-existing domain - :nxdomain
    (db_connection 2.4.2) lib/db_connection/connection.ex:100: DBConnection.Connection.connect/2
    (connection 1.1.0) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 3.17.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: nil
01:54:41.239 [error] GenServer #PID<0.194.0> terminating
** (DBConnection.ConnectionError) tcp connect (plausible_db:5432): non-existing domain - :nxdomain
    (db_connection 2.4.2) lib/db_connection/connection.ex:100: DBConnection.Connection.connect/2
    (connection 1.1.0) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 3.17.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: nil
```
But eventually it seems to succeed
```
01:56:55.209 [error] GenServer #PID<0.194.0> terminating
** (DBConnection.ConnectionError) tcp connect (plausible_db:5432): non-existing domain - :nxdomain
    (db_connection 2.4.2) lib/db_connection/connection.ex:100: DBConnection.Connection.connect/2
    (connection 1.1.0) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 3.17.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: nil
Loading plausible..
Starting dependencies..
Starting repos..
create Plausible.Repo database if it doesn't exist
create Plausible.ClickhouseRepo database if it doesn't exist
Creation of Db successful!
```

**Reply by ruslandoga** [reply]
Where does `Service Unavailable` come from? If it's from a reverse proxy, can you check what Plausible is in when you get that response?
